api/publish.go

```diff
@@ -347,12 +431,28 @@ func apiPublishUpdateSwitch(c *gin.Context) {
 		published.MultiDist = *b.MultiDist
 	}
 
-	resources = append(resources, string(published.Key()))
-	taskName := fmt.Sprintf("Update published %s (%s): %s", published.SourceKind, strings.Join(updatedComponents, " "), strings.Join(updatedSnapshots, ", "))
+	revision := published.ObtainRevision()
+	sources := revision.Sources
+
+	if published.SourceKind == deb.SourceSnapshot {
+		for _, snapshotInfo := range b.Snapshots {
+			component := snapshotInfo.Component
+			name := snapshotInfo.Name
+			sources[component] = name
+		}
+	}
+
+	resources := []string{string(published.Key())}
+	taskName := fmt.Sprintf("Update published %s repository %s/%s", published.SourceKind, published.StoragePrefix(), published.Distribution)
 	maybeRunTaskInBackground(c, taskName, resources, func(out aptly.Progress, _ *task.Detail) (*task.ProcessReturnValue, error) {
-		err := published.Publish(context.PackagePool(), context, collectionFactory, signer, out, b.ForceOverwrite)
+		result, err := published.Update(collectionFactory, out)
```

Comment:
maybe run outside of maybeRunTaskInBackground ? this just updates the data structuresm right ?

Reply:
We can put it outside. I was not sure, what should/must be done outside or inside of a task.

Reply:
was this part of the logic inside or outside of the background task ?

context: the database will be unlocked, until the background tasks runs, and other tasks usind the same prefix will lock the resource. I think everything modifying should be inside the block. maybe that was not the case before, which could lead to race conditions with parallel operations...

Reply:
The data structure modification was done outside of the task before the change.

Reply:
I think doing this inside is actually fixing a race condition, since go routines run in parallel and multiple requests may overwrite the data structures before the background tasks are started (and the resources are locked).